src/main/java/uk/gov/hmcts/juror/api/moj/repository/letter/court/FailedToAttendLetterListRepositoryImpl.java

```diff
@@ -1,80 +1,129 @@
 
         filterEligibleLetterSearchCriteria(jpaQuery, searchCriteria);
 
         orderQueryResults(jpaQuery, searchCriteria.includePrinted());
 
-        return jpaQuery.fetch();
+        List<Tuple> results = jpaQuery.fetch();
+
+        return results.stream()
+            .map(tuple -> FailedToAttendLetterList.builder()
+                .poolNumber(tuple.get(QPoolRequest.poolRequest.poolNumber))
+                .jurorNumber(tuple.get(QJuror.juror.jurorNumber))
+                .firstName(tuple.get(QJuror.juror.firstName))
+                .lastName(tuple.get(QJuror.juror.lastName))
+                .postcode(tuple.get(QJuror.juror.postcode))
+                .status(tuple.get(QJurorStatus.jurorStatus.statusDesc))
+                .absentDate(tuple.get(QAppearance.appearance.attendanceDate))
+                .datePrinted(tuple.get(QJurorHistory.jurorHistory.dateCreated))
+                .isActive(tuple.get(QJurorPool.jurorPool.isActive))
+                .build())
+            .collect(Collectors.toList());
     }
 
 
-    private void filterEligibleLetterSearchCriteria(JPAQuery<FailedToAttendLetterList> jpaQuery,
+    private JPAQuery<Tuple> buildBaseQuery(String owner) {
+
+        JPAQueryFactory queryFactory = new JPAQueryFactory(entityManager);
+
+        return queryFactory.select(
+                QPoolRequest.poolRequest.poolNumber,
+                QJuror.juror.jurorNumber,
+                QJuror.juror.firstName,
+                QJuror.juror.lastName,
+                QJuror.juror.postcode,
+                QJurorStatus.jurorStatus.statusDesc,
+                QJurorHistory.jurorHistory.dateCreated,
+                QJurorPool.jurorPool.isActive,
+                QAppearance.appearance.attendanceDate)
+            .from(QJurorPool.jurorPool)
+            .join(QJuror.juror).on(QJuror.juror.eq(QJurorPool.jurorPool.juror))
+            .join(QPoolRequest.poolRequest).on(QPoolRequest.poolRequest.eq(QJurorPool.jurorPool.pool))
+            .join(QJurorStatus.jurorStatus).on(QJurorStatus.jurorStatus.eq(QJurorPool.jurorPool.status))
+            .join(QAppearance.appearance).on(QAppearance.appearance.jurorNumber.eq(QJuror.juror.jurorNumber))
+            .leftJoin(QJurorHistory.jurorHistory)
+            .on(QJurorHistory.jurorHistory.jurorNumber.eq(QJuror.juror.jurorNumber)
+                .and(QJurorHistory.jurorHistory.poolNumber.eq(QPoolRequest.poolRequest.poolNumber))
+                .and(QJurorHistory.jurorHistory.historyCode.eq(HistoryCodeMod.FAILED_TO_ATTEND_LETTER)))
+            .where(QJurorPool.jurorPool.isActive.eq(Boolean.TRUE)
+                .and(QJurorPool.jurorPool.status.status.in(List.of(IJurorStatus.RESPONDED,
+                        IJurorStatus.FAILED_TO_ATTEND))
+                .and(QAppearance.appearance.noShow.eq(Boolean.TRUE))
+                .and(QAppearance.appearance.attendanceType.eq(AttendanceType.ABSENT))
+                .and(QPoolRequest.poolRequest.owner.eq(owner))));
```

Comment:
Should this check loc code instead of owner?

Reply:
inclined to leave it as is for now as it replicates what was signed off but can see if users want loc_code going forward